### PR TITLE
Troubleshooting for EVH

### PR DESCRIPTION
**Pull request by akshayhavile**

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -213,3 +213,28 @@ No. After changing the serviceType, AKO has to be rebooted and all objects which
...
+- Section `Role-Based Access Control (RBAC)` should show markers associated with the object.
+
+![Alt-text](./images/Markers-on-UI.png)
+
+
+2. Avi Shell
+
+Markers can be viewed for an object with the command.
+
+  `[avi-shell] show <object type> <object name>`
```
Comment (monotosh-avi) [inline docs/faq.md:236]: Would it help if we add a snippet of the relevant output and explain the markers for this example here ?

**Comment by akshayhavile** [comment]
build ako

**Review by monotosh-avi — APPROVED** [review]

File: docs/faq.md [hunk 2]
```diff
@@ -213,3 +213,48 @@ No. After changing the serviceType, AKO has to be rebooted and all objects which
...
 
 No. After changing the serviceType, the users have to remove NPL annotation from the Services themselves.
+
+#### Where can I see markers associated with Avi object?
+
+Markers, associated with an Avi object, are visible on the Avi Vantage UI and the Avi shell.
+
+1. Avi Vantage UI: 
+
+Markers will be visible on UI for avi controller version >= 20.1.6
```
Comment (sudswasavi) [inline docs/faq.md:223]: Markers are visible...

File: docs/faq.md [hunk 3]
```diff
@@ -213,3 +213,48 @@ No. After changing the serviceType, AKO has to be rebooted and all objects which
...
+
+- Edit an object.
```
Comment (sudswasavi) [inline docs/faq.md:225]: In order to view marker labels for an object, do the following:

File: docs/faq.md [hunk 4]
```diff
@@ -213,3 +213,48 @@ No. After changing the serviceType, AKO has to be rebooted and all objects which
 #### If serviceType is changed from NodePortLocal, would AKO remove NPL annotation from the Services automatically ?
 
 No. After changing the serviceType, the users have to remove NPL annotation from the Services themselves.
+
+#### Where can I see markers associated with Avi object?
```
Comment (sudswasavi) [inline docs/faq.md:217]: How can I see the marker labels associated with an object on the Avi Controller?

File: docs/faq.md [hunk 5]
```diff
@@ -213,3 +213,48 @@ No. After changing the serviceType, AKO has to be rebooted and all objects which
...
+
+Markers, associated with an Avi object, are visible on the Avi Vantage UI and the Avi shell.
+
+1. Avi Vantage UI: 
+
+Markers will be visible on UI for avi controller version >= 20.1.6
+
+- Edit an object.
+- Navigate to the `Advanced` tab.
+- Section `Role-Based Access Control (RBAC)` should show markers associated with the object.
```
Comment (sudswasavi) [inline docs/faq.md:227]: The `Role-Based Access Control (RBAC)` shows the markers associated with the object.

File: docs/troubleshooting.md [hunk 6]
```diff
@@ -181,6 +181,11 @@ Sample Output with PVC :
 
 It's recommended we collect the controller tech support logs as well. Please follow this [link](https://avinetworks.com/docs/18.2/collecting-tech-support-logs/)  for the controller tech support.
 
+## Troubleshooting for AKO EVH mode
+### How do I debug an issue in AKO in EVH mode as Avi object names are encoded?
+
+With AKO version 1.5.1 and controller version 20.1.5, AKO marks each Avi object, created by it, with the set of markers. These markers help in identifying objects belong to specific kubernetes or opeshift object. List of markers, associated with each Avi object, can be found out [here](objects.md#markers-for-avi-objects)
```
Comment (sudswasavi) [inline docs/troubleshooting.md:187]: Even though the EVH objects are encoded, AKO labels each EVH object on the controller with a set of key/values that act as metadata for the object. These labels or markers can be used to know, the corresponding kubernetes identifiers for the object.

**Review by sudswasavi — APPROVED** [review]

**Pull request merged**